### PR TITLE
Add analyze-ts method

### LINKED ISSUE #2379: [EslintBridge] Implement analyze-ts method
#### Description

Implement request and response for eslintbridge:analyze-ts.

### PR DESCRIPTION
**Pull request by rita-sonarsource**
Fixes #2379

### CODE REVIEW

File: src/TypeScript/EslintBridgeClient/EslintBridgeClient.cs [hunk 1]
```diff
@@ -128,6 +133,25 @@ public async Task<TSConfigResponse> TsConfigFiles(string tsConfigFilePath, Cance
...
+
+            var responseString = await httpWrapper.PostAsync("analyze-ts", analysisRequest, cancellationToken);
+
+            if (string.IsNullOrEmpty(responseString))
+            {
+                throw new InvalidOperationException(Resources.ERR_InvalidResponse);
+            }
+
+            return JsonConvert.DeserializeObject<AnalysisResponse>(responseString);
+        }
```
Comment (duncanp-sonar) [inline src/TypeScript/EslintBridgeClient/EslintBridgeClient.cs:153]: Apart from the `tsconfigFilePath`, this method is identical to the `AnalyzeJs` method one, isn't it? If so, it seems like a good candidate for factoring out most of the behaviour into a helper method.

File: src/TypeScript.UnitTests/EslintBridgeClient/EslintBridgeClientTests.cs [hunk 2]
```diff
@@ -207,6 +207,78 @@ public async Task TSConfigFiles_HasResponse_DeserializedResponse()
             result.Should().BeEquivalentTo(response);
         }
 
+        [TestMethod]
```
Comment (duncanp-sonar) [inline src/TypeScript.UnitTests/EslintBridgeClient/EslintBridgeClientTests.cs:210]: I'm guessing these test are copies of the JS ones with js->ts changes?

**Review by duncanp-sonar — APPROVED** [review]
Looks good. Just the comment about code duplication.

File: src/TypeScript.UnitTests/EslintBridgeClient/EslintBridgeClientTests.cs (hunk 2 above)
Comment (rita-sonarsource) [reply src/TypeScript.UnitTests/EslintBridgeClient/EslintBridgeClientTests.cs:210]: Yep; sorry I should've said that.

**Pull request merged**